includes/classes/Feature/MetaKeyPattern.php

```diff
@@ -85,54 +85,87 @@ public function setup() {
+					</p>
+				</div>
 			</div>
-			<div class="input-wrap">
-				<textarea
-					class="setting-field large-text code"
-					id="meta_key_allow_pattern"
-					rows="4"
-					name="settings[meta_key_allow_pattern]"
-				><?php echo empty( $settings['meta_key_allow_pattern'] ) ? '' : esc_textarea( $settings['meta_key_allow_pattern'] ); ?></textarea>
-				<p class="field-description">
-					<?php esc_html_e( 'Separate multiple regular expressions with line breaks.', 'elasticpress-labs' ); ?>
-					<?php esc_html_e( 'Include the weight of the pattern adding a pipe (|) followed by a number. Example: /^[a-z]/|5', 'elasticpress-labs' ); ?>
-				</p>
-			</div>
-		</div>
 
-		<div class="field js-toggle-feature" data-feature="<?php echo esc_attr( $this->slug ); ?>">
-			<div class="field-name status">
-				<label for="meta_key_deny_pattern">
-					<?php esc_html_e( 'Deny patterns', 'elasticpress-labs' ); ?>
-				</label>
-			</div>
-			<div class="input-wrap">
-				<textarea
-					class="setting-field large-text code"
-					id="meta_key_deny_pattern"
-					rows="4"
-					name="settings[meta_key_deny_pattern]"
-				><?php echo empty( $settings['meta_key_deny_pattern'] ) ? '' : esc_textarea( $settings['meta_key_deny_pattern'] ); ?></textarea>
-				<p class="field-description">
-					<?php esc_html_e( 'Separate multiple regular expressions with line breaks.', 'elasticpress-labs' ); ?>
-				</p>
+			<div class="field js-toggle-feature" data-feature="<?php echo esc_attr( $this->slug ); ?>">
+				<div class="field-name status">
+					<label for="meta_key_deny_pattern">
+						<?php esc_html_e( 'Deny patterns', 'elasticpress-labs' ); ?>
+					</label>
+				</div>
+				<div class="input-wrap">
+					<textarea
+						class="setting-field large-text code"
+						id="meta_key_deny_pattern"
+						rows="4"
+						name="settings[meta_key_deny_pattern]"
+					><?php echo empty( $settings['meta_key_deny_pattern'] ) ? '' : esc_textarea( $settings['meta_key_deny_pattern'] ); ?></textarea>
+					<p class="field-description">
+						<?php esc_html_e( 'Separate multiple regular expressions with line breaks.', 'elasticpress-labs' ); ?>
+					</p>
+				</div>
 			</div>
-		</div>
-		<?php
+			<?php
+			return;
+		}
+
+		_doing_it_wrong(
+			__METHOD__,
+			esc_html__( 'Settings are now generated via the set_settings_schema() method.' ),
+			'ElasticPress Labs 2.2.0'
+		);
+	}
+
+	/**
+	 * Set the `settings_schema` attribute
+	 *
+	 * @since 2.2.0
+	 */
+	public function set_settings_schema() {
+		$this->settings_schema = [
+			[
+				'default' => '',
+				'help'    => __( '<p>Separate multiple regular expressions with line breaks.</p><p>Include the weight of the pattern adding a pipe (|) followed by a number. Example: /^[a-z]/|5</p>', 'elasticpress-labs' ),
+				'key'     => 'meta_key_allow_pattern',
+				'label'   => __( 'Allow patterns', 'elasticpress-labs' ),
+				'type'    => 'textarea',
```

Comment:
@felipeelia The main plugin doesn't currently support `textarea`. That will need to be added before this works.